src/Promitor.Core.Scraping/ResourceTypes/DataShareScraper.cs

```diff
@@ -38,26 +37,31 @@ protected override string DetermineMetricFilter(string metricName, DataShareReso
             return $"{fieldName} eq '{entityName}'";
         }
 
-        protected override string DetermineMetricDimension(string metricName, DataShareResourceDefinition resourceDefinition, MetricDimension dimension)
+        protected override List<string> DetermineMetricDimensions(string metricName, DataShareResourceDefinition resourceDefinition, List<MetricDimension> dimensions)
         {
             if (IsShareNameConfigured(resourceDefinition))
             {
-                return base.DetermineMetricDimension(metricName, resourceDefinition, dimension);
+                return base.DetermineMetricDimensions(metricName, resourceDefinition, dimensions);
             }
 
             var dimensionName = GetMetricFilterFieldName(metricName);
             Logger.LogTrace($"Using '{dimensionName}' dimension since no share name was configured.");
 
-            return dimensionName;
+            return new List<string> { dimensionName };
         }
 
-        protected override List<MeasuredMetric> EnrichMeasuredMetrics(DataShareResourceDefinition resourceDefinition, string dimensionName, List<MeasuredMetric> metricValues)
+        protected override List<MeasuredMetric> EnrichMeasuredMetrics(DataShareResourceDefinition resourceDefinition, List<string> dimensionNames, List<MeasuredMetric> metricValues)
         {
             // Change Azure Monitor dimension name to more representable value
-            foreach (var measuredMetric in metricValues.Where(metricValue => metricValue.DimensionName == "ShareName"
-                                                                             || metricValue.DimensionName == "ShareSubscriptionName"))
+            foreach (var measuredMetric in metricValues)
             {
-                measuredMetric.DimensionName = "share_name";
+                for (var i = 0; i < measuredMetric.DimensionNames.Count; i++)
+                {
+                    if (measuredMetric.DimensionNames[i] == "ShareName" || measuredMetric.DimensionNames[i] == "ShareSubscriptionName")
```

Comment:
The problem with a `where` in that version was, that as far as I understand you can not modify the results from the `where` directly. With the new `MeasuredMetricDimension` this problem is however solved, since the `where` selects only the `MeasuredMetricDimension`, so we can modify its members.

Reply:
I think a where with foreach should work though 🤔 I'd give it a try to simplify things. Can you give that a try please?

Reply:
It actually already is a `Where` again now, since the other change fixed the issue with not being able to modify the name :)